File: Arrays/Equilibrium_Point.cpp
class Solution {
  public:

    int equilibriumPoint(long long arr[], int n) {
        if(n==1)return 1;
        long long sum = 0;
        for(int i=0;i<n;i++){
            sum+= arr[i];
        }
        long long left = 0,right = 0;
        for(int i = 0;i<n;i++){
            right = sum - left - arr[i];
            if(left == right){return i+1;}
            left+= arr[i];
        }
        return -1;
    }
};

File: Arrays/Equilibrium_Point.py
class Solution:
    # Complete this function
    
    #Function to find equilibrium point in the array.
    def equilibriumPoint(self,arr, n):
        # Your code here
        sum = 0
        for i in range(n):
            sum+=arr[i]
        left,right = 0,0
        
        for i in range(n):
            right = sum - left - arr[i]
            if left == right:
                return i+1
            left+=arr[i]
        return -1
